### PR TITLE
779 game table improvements

### PR DESCRIPTION
**Pull request by chao-mu**

### CODE REVIEW

File: frontend/src/components/games/list/GameListItem.tsx [hunk 1]
```diff
@@ -153,6 +235,38 @@ export function RenderResult(params: GridRenderCellParams) {
     );
 }
 
+function parseTimeControl(
```
Comment (jackstenglein) [inline frontend/src/components/games/list/GameListItem.tsx:238]: Parsing a time control is actually more complicated than this, but we can probably use this for the majority of games that are being submitted by users. There's probably some master games where this won't be accurate, but not a big deal

File: frontend/src/components/games/list/GameListItem.tsx [hunk 2]
```diff
@@ -153,6 +235,38 @@ export function RenderResult(params: GridRenderCellParams) {
...
+    return [Math.round(gameTime / 60), increment];
+}
+
+export function getTimeControl({ timeControl }: { timeControl?: string }) {
+    const [gameTime, increment] = parseTimeControl(timeControl);
+
+    return gameTime ? `${gameTime}+${increment ?? 0}` : '';
+}
+
+export function RenderTimeControl({ timeControl }: { timeControl?: string }) {
```
Comment (jackstenglein) [inline frontend/src/components/games/list/GameListItem.tsx:262]: Should this return null if timeControl is undefined/empty rather than rendering an empty div/p?

File: frontend/src/components/games/list/GameTable.tsx [hunk 3]
```diff
@@ -18,115 +18,110 @@ import {
...
         valueGetter: (_value, row) =>
             `${row.headers.White} (${row.headers.WhiteElo}) - ${row.headers.Black} (${row.headers.BlackElo})`,
         renderCell: RenderPlayersCell,
-        flex: 1,
-        minWidth: 150,
+        minWidth: 200,
+        flex: 2,
     },
     {
-        field: 'result',
```
Comment (jackstenglein) [inline frontend/src/components/games/list/GameTable.tsx:49]: One side effect of removing this column is that we lose the ability to sort by result, which is a rather big loss

File: frontend/src/components/ui/CardGrid.tsx
Comment (jackstenglein) [inline frontend/src/components/ui/CardGrid.tsx:1]: Did I miss it somewhere or is this file not used?

File: frontend/src/components/games/list/GameCard.tsx
Comment (jackstenglein) [inline frontend/src/components/games/list/GameCard.tsx:1]: Did I miss it somewhere or is this file not used?

File: frontend/src/components/games/list/GameListItem.tsx (hunk 1 above)
Comment (chao-mu) [reply frontend/src/components/games/list/GameListItem.tsx:238]: Do we have examples of the logic elsewhere? I know there are multiple time increments... The logic here is specifically for Lichess and I'm wondering if I should check that and if it's not Lichess use verbatim?

Comment (jackstenglein) [reply frontend/src/components/games/list/GameListItem.tsx:266]: You can look at the Clock Usage underboard tab.

Also, a full description of the TimeControl header is here: http://www.saremba.de/chessgml/standards/pgn/pgn-complete.htm#c9.6.1

Again, the vast majority of games will have a simple time control header, so this is pretty low priority

File: frontend/src/components/games/list/GameTable.tsx [hunk 4]
```diff
@@ -91,9 +149,16 @@ export default function GameTable({
     contextMenu,
     limitFreeTier,
     columns,
+    defaultVisibility,
 }: GameTableProps) {
     const freeTierLimited = useFreeTier() && limitFreeTier;
     const { data, request, page, pageSize, rowCount, hasMore, setPage } = pagination;
+    const [columnVisibility, setColumnVisibility] = useState<GridColumnVisibilityModel>({
```
Comment (jackstenglein) [inline frontend/src/components/games/list/GameTable.tsx:156]: If you're wanting to make these settings sticky, you can do it very easily by replacing the useState call with useLocalStorage

File: frontend/src/components/games/list/GameListItem.tsx (hunk 1 above)
Comment (chao-mu) [reply frontend/src/components/games/list/GameListItem.tsx]: Okay, I didn't mess too much with it, but looking at ClockUsage showed me there was existing functions I could use so I dropped them in.

File: frontend/src/components/games/list/GameTable.tsx (hunk 3 above)
Comment (chao-mu) [reply frontend/src/components/games/list/GameTable.tsx:49]: Was added back in.

File: frontend/src/components/games/list/GameTable.tsx (hunk 4 above)
Comment (chao-mu) [reply frontend/src/components/games/list/GameTable.tsx]: Done. Added a namespace since different pages will want different columns visible (e.g. your own games versus other's)

**Comment by chao-mu** [comment]
Ready for final code review.

File: frontend/src/profile/GamesTab.tsx [hunk 5]
```diff
@@ -20,6 +20,7 @@ interface GamesTabProps {
 }
 
 const GamesTab: React.FC<GamesTabProps> = ({ user }) => {
+    const navigate = useNavigate();
```
Comment (jackstenglein) [inline frontend/src/profile/GamesTab.tsx:23]: Can you revert these changes related to useNavigate()? It breaks the headers NextJS has to return to enable Stockfish. Also, we're eventually going to stop using react-router-dom and use the NextJS app router instead

Comment (chao-mu) [reply frontend/src/profile/GamesTab.tsx:23]: What should I do instead? These are existing changes
